editoast/src/views/rolling_stocks.rs

```diff
@@ -0,0 +1,64 @@
+use crate::error::Result;
+use crate::schema::rolling_stock::RollingStock;
+use crate::DbPool;
+use actix_web::dev::HttpServiceFactory;
+use actix_web::get;
+use actix_web::web::{self, block, Data, Json, Path};
+
+pub fn routes() -> impl HttpServiceFactory {
+    web::scope("/rolling_stock").service(get)
+}
+
+/// Return a specific set of electrical profiles
+#[get("/{rolling_stock}")]
+async fn get(db_pool: Data<DbPool>, rolling_stock: Path<i64>) -> Result<Json<RollingStock>> {
+    let rolling_stock = rolling_stock.into_inner();
+    block(move || Ok(Json(RollingStock::retrieve(db_pool, rolling_stock)?)))
```

Comment:
The block should be done in the `RollingStock::retrieve` function just before you take a DB connection.